src/main/java/cn/nukkit/block/BlockDoor.java

```diff
@@ -293,6 +293,7 @@ public boolean toggle(Player player) {
         this.getLevel().getServer().getPluginManager().callEvent(event);
 
         if (event.isCancelled()) {
+            getLevel().sendBlocks(new Player[] {player}, new Block[] {this.down()}); // fixes a client-prediction bug
```

Comment:
If this only affects the upper part of the door, the code should check for that otherwise it's going to update the ground below the door when the lower part is pressed.

Reply:
@SupremeMortal I know, anyway this fixes it. Updating only the upper half is leading to nothing.